### ISSUE TITLE
Things to do to make the job cache more useful

### ISSUE THREAD

**Issue by mvdbeek** [galaxyproject/galaxy#6887]
- [ ] Datasets imported from a library are not recognized as being the same
- [x] The input and output datasets need to be owned by the user -- accessible should be enough (though that might be  a concern with our accessible by default policy)
- [ ] #6781 having the HID as part of the name really requires one to have the exact same history structure
- [ ] A tool property that says "ignore the element_identifier / name when caching" would be helpful
- [ ] A tool property to indicate the tool outputs are not reproducible (e.g. tools that produce random results on purpose)
- [ ] An additional setting that would allow the user to specify that an exact metadata match isn't necessary
- [ ] Enable cache override via UI element in the tool form.
- [ ] A preview
- [ ] Make datasets read-only in object store https://github.com/galaxyproject/galaxy/issues/8263
- [ ] If we have a hash for dataset contents we can "resume" cached execution when intermediate files get deleted or when the dataset metadata for an input file didn't match but the metadata mismatch didn't result in a different output
- [x] Add something to the UI to indicate that a job was recovered from cached data
- [ ] https://github.com/galaxyproject/galaxy/issues/20050

**Comment by hexylena** [comment]
ping @wm75 who had similar complaints when he tried it yesterday.

**Comment by wm75** [comment]
Agree on all of the above. Plus, I'd add:
A more informative status message of what's happening. In my hands whenever the server was using a cached result, after clicking Execute the page appeared to be frozen sending the job.
It was only after several minutes that it switched to job submission confirmation. At the same time, the new datasets suddenly appeared in my history.
I can only speculate, but maybe that's because of the large size of Galaxy EU's database?

**Comment by mvdbeek** [comment]
I don't think database size should have a major effect. How many datasets were you generating there ?

**Comment by wm75** [comment]
per job? Between one and four.
Or are you asking how many datasets I own on the server (not sure about that, but maybe 2000)?

**Comment by mvdbeek** [comment]
No, the number of items a tool execution would produce. So yeah, that is less than ideal ...

**Comment by lecorguille** [comment]
If you shared a history for training purpose, perhaps, this feature can be useful, isn't it?
-> Not from library, same history structure

But maybe we hit this wall?
> The input and output datasets need to be owned by the user

since the datasets and the cached jobs are owned by the instructor?

**Comment by bwlang** [comment]
Would be good to add something to the UI to indicate that a job was recovered from cached data.  Also, would be cool to enable cache override via UI element in tool form

**Comment by bwlang** [comment]
I like these approaches.  Job caching could have a huge impact in real-world scenarios (and is starting to in our local galaxy).  I'm happy hear that you're still on it!

**Comment by hexylena** [comment]
Xref our discussion in matrix, here was a small reproduction

- [Workflow](https://usegalaxy.eu/u/helena-rasche/w/testing-job-caching-reuse)
- [History](https://usegalaxy.eu/u/helena-rasche/h/job-caching-testing-2)

I ran two invocations, the first without caching ([`51b7692a0622cfe4`](https://usegalaxy.eu/api/invocations/51b7692a0622cfe4)), the second with ([`a7f9ca99b7729f0f`](https://usegalaxy.eu/api/invocations/a7f9ca99b7729f0f)). can provide more information from the api later if it's useful.

**Comment by mvdbeek** [comment]
> * [Getting rid of the hid ? #6781](https://github.com/galaxyproject/galaxy/issues/6781) having the HID as part of the name really requires one to have the exact same history structure

https://github.com/galaxyproject/galaxy/pull/19962 is a 90% solution here, I think.
